src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -933,10 +934,35 @@ FixedwingPositionControl::control_position(const math::Vector<2> &curr_pos, cons
 			_att_sp.yaw_body = 0;
 		}
 
+	} else if (_control_mode.flag_control_altitude_enabled
+		   && _control_mode.flag_control_offboard_enabled
+		   && pos_sp_curr.valid) {
+		/* Offboard altitude mode, control on yaw rate with fixed altitude */
+		_control_mode_current = FW_POSCTRL_MODE_OFFBOARD_ALTITUDE;
+
+		_hold_alt = _global_pos.alt;
+		float airspeed = _parameters.airspeed_trim;
+		float throttle = _parameters.throttle_cruise;
+		float gnd_speed = ground_speed.length();
+		tecs_update_pitch_throttle(pos_sp_curr.z,
+					   calculate_target_airspeed(airspeed),
+					   radians(_parameters.pitch_limit_min) - _parameters.pitchsp_offset_rad,
+					   radians(_parameters.pitch_limit_max) - _parameters.pitchsp_offset_rad,
+					   _parameters.throttle_min,
+					   _parameters.throttle_max,
+					   throttle,
+					   false,
+					   radians(_parameters.pitch_limit_min));
+
+		// Calculate the bank angle needed to achieve the yaw rate
+		float roll_sp = atan2f(pos_sp_curr.yawspeed * gnd_speed, 9.8065f);
```

Comment:
You can use this constant for gravity. https://github.com/PX4/Firmware/blob/master/src/lib/geo/geo.h#L53